lib/replica_pools/pools.rb

```diff
@@ -53,17 +53,18 @@ def config_hash
     # generates a unique ActiveRecord::Base subclass for a single replica
     def connection_class(pool_name, replica_name, connection_name)
       class_name = "#{pool_name.camelize}#{replica_name.camelize}"
-
+      connection_config_method_name = ReplicaPools::ConnectionProxy.get_connection_config_method_name
       ReplicaPools.const_set(class_name, Class.new(ActiveRecord::Base) do |c|
         c.abstract_class = true
-        c.define_singleton_method(:connection_config) do
-          configurations.configs_for(connection_name.to_s)
+        c.define_singleton_method(connection_config_method_name) do
+          configurations.configs_for(env_name: connection_name.to_s, include_hidden: true)
```

Comment:
This behaves differently due to the update of the proxy from `connection_config` to `connection_db_config`. [Here is a concise](https://api.rubyonrails.org/v6.1/classes/ActiveRecord/ConnectionHandling.html#method-i-connection_config) look at what the differences are.

**Important**
`configs_for` takes options instead of any positional arguments, as specified [here](https://api.rubyonrails.org/classes/ActiveRecord/DatabaseConfigurations.html#method-i-configs_for). Did I specify the correct options? Here is an example of the output when running the tests locally and pry'ing in:
```

    55:     def connection_class(pool_name, replica_name, connection_name)
    56:       class_name = "#{pool_name.camelize}#{replica_name.camelize}"
    57:       connection_config_method_name = ReplicaPools::ConnectionProxy.get_connection_config_method_name
    58:       ReplicaPools.const_set(class_name, Class.new(ActiveRecord::Base) do |c|
    59:         c.abstract_class = true
 => 60:         binding.pry
    61:         c.define_singleton_method(connection_config_method_name) do
    62:           configurations.configs_for(env_name: connection_name.to_s, include_hidden: true)
    63:         end
    64:       end)
    65:

[1] pry(#<Class>)> connection_name
=> "test_pool_secondary_name_db1"
[2] pry(#<Class>)> configurations.configs_for(env_name: connection_name.to_s, include_hidden: true)
=> [#<ActiveRecord::DatabaseConfigurations::HashConfig:0x00007f84a093d948
  @configuration_hash=
   {:adapter=>"mysql2",
    :username=>"read_only",
    :password=>"readme",
    :host=>"127.0.0.1",
    :port=>3309,
    :encoding=>"utf8",
    :read_timeout=>1,
    :database=>"test_db"},
  @env_name="test_pool_secondary_name_db1",
  @name="primary">]
```
My assumption is that we want an individual config. If we pass no `env_name` or `name` argument, this `configurations.configs_for` returns an Array of `ActiveRecord::DatabaseConfigurations::HashConfig` which doesn't seem suitable for generating a subclass for a **single** replica.

Please do double check my assumptions though! @talaris your input would be greatly appreciated here.